docs/docs/ref-06-dom-differences.md

```diff
@@ -10,6 +10,7 @@ React has implemented a browser-independent events and DOM system for performanc
 
 * All DOM properties and attributes (including event handlers) should be camelCased to be consistent with standard JavaScript style. We intentionally break with the spec here since the spec is inconsistent. **However**, `data-*` and `aria-*` attributes [conform to the specs](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes#data-*) and should be lower-cased only.
 * The `style` attribute accepts a JavaScript object with camelCased properties rather than a CSS string. This is consistent with the DOM `style` JavaScript property, is more efficient, and prevents XSS security holes.
+* Since `class` and `for` are reserved words in JavaScript, the JSX elements for built-in [DOM nodes](http://javascript.info/tutorial/dom-nodes) should use the attribute names `className` and `htmlFor` respectively, (eg. `<div className="foo" />` ). Custom elements should use `class` and `for` directly (eg. `<my-tag class="foo" />` ).
```

Comment:
Super delayed response from me, sorry. I need to do better with my email management…

A couple comments here that I'll probably followup with a PR for.
1. Being reserved words was definitely a consideration at the time (could always transform to string keys), but was not the primary reason. The main reason was that className and htmlFor aligned with the JS API. In hindsight… well it was probably a mistake but I think we're probably too far gone down this path.
2. I'd rather link to something a bit more known, so I'm going to link to MDN - https://developer.mozilla.org/en-US/docs/Web/HTML/Element
3. I'll probably add a tiny addendum about the fact that DOM APIs don't make sense for custom elements, so we pass all attributes along directly.